core/os/thread.h

```diff
@@ -29,19 +29,30 @@
 /**************************************************************************/
 
 #include "platform_config.h"
+
 // Define PLATFORM_THREAD_OVERRIDE in your platform's `platform_config.h`
-// to use a custom Thread implementation defined in `platform/[your_platform]/platform_thread.h`
-// Overriding the platform implementation is required in some proprietary platforms
+// to use a custom Thread implementation defined in `platform/[your_platform]/platform_thread.h`.
+// Overriding the Thread implementation is required in some proprietary platforms.
+
 #ifdef PLATFORM_THREAD_OVERRIDE
+
 #include "platform_thread.h"
+
 #else
 
 #ifndef THREAD_H
 #define THREAD_H
 
-#include "core/templates/safe_refcount.h"
 #include "core/typedefs.h"
 
+#ifdef THREADS_ENABLED
+
+#include "core/templates/safe_refcount.h"
+
+#if defined(__cpp_lib_hardware_interference_size) && !defined(ANDROID_ENABLED)
+#include <new> // For CACHE_LINE_BYTES.
+#endif
```

Comment:
I'm assuming that adding the `#if` conditions matching where it's used will prevent tools like clangd to iwyu from flagging this as unused when run on platforms which don't fulfill that condition (seems to be GCC specific, possibly supported on Clang, but thus not MSVC most likely).

If not, we should add a pragma comment to prevent its removal, but then I wouldn't trust the judgement of those tools.

Reply:
Ok this is what broke my fix, to test `__cpp_lib_hardware_interference_size` we first need to include `<new>`.

So pragma comment it will be.